src/lifx.d.ts

```diff
@@ -1,367 +1,328 @@
 
-  /**
-  * Adds a message handler that calls a function when the requested
-  * info was received
-  * @param {String} type A type of packet to listen for, like stateLight
-  * @param {Function} callback the function to call if the packet was received,
-  *                   this will be called with parameters msg and rinfo
-  * @param {Number} [sequenceNumber] Expects a specific sequenceNumber on which will
-  *                                  be called, this will call it only once. If not
-  *                                  given the callback handler is permanent
-  */
-  addMessageHandler(type: any, callback: any, sequenceNumber: any): void;
+  on(event: 'error', callback: (err: Error) => void): this;
+  on(event: 'message', callback: (msg: any, info: RemoteInfo) => void): this;
+  on(event: 'listening', callback: () => void): this;
+  on(event: 'light-offline', callback: (info: Light) => void): this;
+  on(event: 'light-online', callback: (info: Light) => void): this;
+  on(event: 'light-new', callback: (info: Light) => void): this;
+  on(event: 'discovery-completed', callback: () => void): this;
 
-  /**
-  * Get network address data from connection
-  * @return {Object} Network address data
-  */
-  address(): any;
+  /** Get network address data from connection */
+  address(): AddressInfo;
 
-  /**
-  * Destroy an instance
-  */
+  /** Close client */
   destroy(): void;
 
-  /**
-  * Creates a new socket and starts discovery
-  * @example
-  * init({debug: true}, function() {
-  *   console.log('Client started');
-  * })
-  * @param {Object} [options] Configuration to use
-  * @param {String} [options.address] The IPv4 address to bind to
-  * @param {Number} [options.port] The port to bind to
-  * @param {Boolean} [options.debug] Show debug output
-  * @param {Number} [options.lightOfflineTolerance] If light hasn't answered for amount of discoveries it is set offline
-  * @param {Number} [options.messageHandlerTimeout] Message handlers not called will be removed after this delay in ms
-  * @param {String} [options.source] The source to send to light, must be 8 chars lowercase or digit
-  * @param {Boolean} [options.startDiscovery] Weather to start discovery after initialization or not
-  * @param {Array} [options.lights] Pre set list of ip addresses of known addressable lights
-  * @param {String} [options.broadcast] The broadcast address to use for light discovery
-  * @param {Function} [callback] Called after initialation
-  */
-  init(options: any, callback: any): any;
+  /** Creates a new socket and starts discovery */
+  init(options: ClientOptions, callback: () => void): void;
 
-  /**
-  * Find a light by label, id or ip
-  * @param {String} identifier label, id or ip to search for
-  * @return {Object|Boolean} the light object or false if not found
+  /** Find a light
+  * @param identifier label, id or ip to search for
   */
-  light(identifier: any): any;
+  light(identifier: string): Light | false;
```

Comment:
Please revert the JSDoc as it was. It's simpler to have the same text here as in the code. The types in JSDocs are unnecessary though, so you're welcome to remove them if you wish.